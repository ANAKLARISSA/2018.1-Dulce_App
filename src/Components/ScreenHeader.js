import React from 'react';
import {Header, Left, Body, Button, Icon, Title} from 'native-base';
import {StatusBar} from 'react-native';

const styles = {
  header: {
    backgroundColor: '#5f4b8b'
  },
  text: {
    fontFamily: 'Raleway'
  }
};

class ScreenHeader extends React.Component {
  render() {
    return (
        <Header style={styles.header}>
        <StatusBar
            backgroundColor='#5f4b79'
        />
          <Body>
            <Title style={styles.text} > {this.props.title} </Title>
          </Body>
        </Header>
    );
  }
}

export default ScreenHeader;
