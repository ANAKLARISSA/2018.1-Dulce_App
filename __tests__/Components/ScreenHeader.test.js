import React from 'react';
import ScreenHeader from '../../src/Components/ScreenHeader';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
require('bezier');

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer
    .create(<ScreenHeader />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Should press the button', () => {
  const goBack = jest.fn();
  const wrapper = shallow(<ScreenHeader title = 'test' goBack = {() => goBack()}/>);
  const button = wrapper.find('Styled(Button)').at(0);
  button.simulate('press');
  expect(goBack).toHaveBeenCalled();
});
