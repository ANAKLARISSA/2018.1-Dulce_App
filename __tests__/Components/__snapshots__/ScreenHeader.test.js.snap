// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        },
        Object {
          "backgroundColor": "#5f4b8b",
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Raleway",
            },
          ]
        }
        virtual={undefined}
      >
         
         
      </Text>
    </View>
  </View>
</View>
`;
